mantidproject/mantid
# Reorganise unit tests of fitting functions:: 'General' Category - Chunk 1

**Description of work.**

Unit tests mentioned in #16483 which are using `Fit` algorithm from General Category have been moved to `FitTest.h`. Most of the files mentioned in the issue were not utilising the `Fit` algorithm. 

**To test:**
Check all build pass
New unit test implemented for `ExpDecay`
Check that the following files are not use the `Fit` algorithm
- CompositeFunction
- ComptonPeakProfile 
- ComptonScatteringCountRate 
- CrystalFieldSpectrum 
- GaussianComptonProfile 
- GramCharlierComptonProfile 



_Does not need to be in the release notes._
Fixes #16483 



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
